lib/mix/test/mix/tasks/compile.elixir_test.exs

```diff
@@ -82,6 +82,32 @@ defmodule Mix.Tasks.Compile.ElixirTest do
     end)
   end
 
+  test "recompiles project if working directory changed" do
+    in_fixture("no_mixfile", fn ->
+      Mix.Project.push(MixTest.Case.Sample)
+
+      assert Mix.Tasks.Compile.Elixir.run(["--verbose"]) == {:ok, []}
+      assert_received {:mix_shell, :info, ["Compiled lib/a.ex"]}
+      assert_received {:mix_shell, :info, ["Compiled lib/b.ex"]}
+
+      tmp = tmp_path("copied")
+      File.cp_r!(File.cwd!(), tmp)
+      File.cd!(tmp)
+
+      File.touch!("_build/dev/lib/sample/.mix/compile.elixir", @old_time)
+
+      Code.put_compiler_option(:ignore_module_conflict, true)
```

Comment:
I struggled to test this, I have no clear idea if my strategy is good.

Without this I would be getting:
```
warning: redefining module A (current version loaded from /Users/sabiwara/.../elixir/lib/mix/tmp/Mix.Tasks.Compile.ElixirTest/test recompiles project if working directory changed/_build/dev/lib/sample/ebin/Elixir.A.beam)
  lib/a.ex:1: A
```

Even using `purge([A, B])` didn't remove this warning, my mix / `:code` knowledge is not enough here.

Reply:
Honestly, I don’t think a test is necessary. We already test the cache and recompilation elsewhere, which are the important bits.

Reply:
Got it, will remove the test.
At least it got me confident that the fix was working as expected 🙂